jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java

```diff
@@ -0,0 +1,203 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.webapp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.DefaultHandler;
+import org.eclipse.jetty.server.handler.HandlerList;
+import org.eclipse.jetty.toolchain.test.FS;
+import org.eclipse.jetty.toolchain.test.MavenTestingUtils;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.resource.PathResource;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class HugeResourceTest
+{
+    private static final long KB = 1024;
+    private static final long MB = 1024 * KB;
+    private static final long GB = 1024 * MB;
+    public static Path hugeStaticBase;
+    public static Path outputDir;
+
+    public Server server;
+
+    @BeforeAll
+    public static void prepareStaticFiles() throws IOException
+    {
+        hugeStaticBase = MavenTestingUtils.getTargetTestingPath(HugeResourceTest.class.getSimpleName() + "-huge-static-base");
+        FS.ensureDirExists(hugeStaticBase);
+
+        makeStaticFile(hugeStaticBase.resolve("test-1g.dat"), 1 * GB);
+        makeStaticFile(hugeStaticBase.resolve("test-4g.dat"), 4 * GB);
+        makeStaticFile(hugeStaticBase.resolve("test-10g.dat"), 10 * GB);
+
+        outputDir = MavenTestingUtils.getTargetTestingPath(HugeResourceTest.class.getSimpleName() + "-huge-static-outputdir");
+        FS.ensureEmpty(outputDir);
+    }
+
+    @AfterAll
+    public static void cleanupHugeStaticFiles()
```

Comment:
Remove "huge" everywhere, it's not really needed.

Reply:
done.